cppwinrt/code_writers.h

```diff
@@ -575,7 +575,7 @@ namespace cppwinrt
                     w.write(param_name);
                     break;
                 case param_category::array_type:
-                    w.write("%.size(), get_abi(%)", param_name, param_name);
+                    w.write("%.size(), %.size() ? get_abi(%) : (decltype(get_abi(%)))&%", param_name, param_name, param_name, param_name, param_name);
```

Comment:
Would it not be cleaner to update `get_abi` to return a non-null value when the array is empty? Then the logic is codified in one place rather than everywhere `get_abi` is used.

Reply:
For maximum compatibility, I limited the scope of the change to just the `consume_` paths. If you're okay with changing the behavior of `get_abi` globally, then go for it. I don't know what else `get_abi` is used for and didn't want to mess them up accidentally.

Reply:
So the idea is that `get_abi` returns the ABI representation for the projected type. If that is a non-null value for arrays then `get_abi` should ensure that it faithfully returns a null-null value.